.github/workflows/build_test.yml

```diff
@@ -7,6 +7,30 @@ on:
     branches: [main]
 
 jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check License Header
+        uses: apache/skywalking-eyes/header@main
+        with:
+          header:
+            license:
+              spdx-id: Apache-2.0
+      - name: Check CMake files
+        id: cmake-format
+        uses: PuneetMatharu/cmake-format-lint-action@v1.0.0
+        with:
+          args: --config-files .cmake-format.json --in-place
+      - name: Review with Clan-tidy
+        uses: ZedThree/clang-tidy-review@v0.10.0
```

Comment:
This project appears to create a pull request review instead of a failing check?  Though maybe I am misunderstanding their github.  It's also a somewhat small project, though larger than the cmake-format one.

Nevermind, I see it fails later on if there were any comments.

This feels a little over complex.  Maybe we can get away with `run` and a call to `run-clang-tidy`?

Reply:
Updated